src/cat-data-service/src/service/api/registration/mod.rs

```diff
@@ -0,0 +1,203 @@
+use crate::service::common::objects::{event_id::EventId, voter::Voter, voting_key::VotingKey};
+use crate::service::common::responses::resp_2xx::OK;
+use crate::service::common::responses::resp_4xx::NotFound;
+use crate::service::common::responses::resp_5xx::{server_error, ServerError};
+use crate::{service::common::tags::ApiTags, state::State};
+use poem::web::Data;
+use poem_extensions::response;
+use poem_extensions::UniResponse::{T200, T404, T500};
+use poem_openapi::{
+    param::{Path, Query},
+    payload::Json,
+    OpenApi,
+};
+use std::sync::Arc;
+
+pub(crate) struct RegistrationApi;
+
+#[OpenApi(prefix_path = "/registration", tag = "ApiTags::Registration")]
+impl RegistrationApi {
+    #[oai(
+        path = "/voter/:voting_key",
+        method = "get",
+        operation_id = "getVoterInfo"
+    )]
+    /// Voter's info
+    ///
+    /// Get voter's registration and voting power by their voting key.
+    /// If the `event_id` query parameter is omitted, then the latest voting power is retrieved.
+    /// If the `with_delegators` query parameter is ommitted, then `delegator_addresses` field of `VoterInfo` type does not provided.
+    ///
+    async fn get_voter_info(
+        &self,
+        pool: Data<&Arc<State>>,
+
+        /// A Voting Key.
+        #[oai(validator(max_length = 64, min_length = 64, pattern = "[0-9a-f]{64}"))]
+        voting_key: Path<VotingKey>,
+
+        /// The Event ID to return results for.
```

Comment:
```suggestion
        /// The Event ID to return results for.
        /// See [GET Events](Link to events endpoint) for details on retrieving all valid event IDs.
```
This is an example, but we should be cross-linking our documentation as best we can.

Reply:
Sure, after I will add events types and endpoint I will update this place